torch/autograd/functional.py

```diff
@@ -250,26 +250,31 @@ def vjp(func, inputs, v=None, create_graph=False, strict=False):
          (tensor([2., 2.]), tensor([3., 3.])))
     """
 
-    is_inputs_tuple, inputs = _as_tuple(inputs, "inputs", "vjp")
-    inputs = _grad_preprocess(inputs, create_graph=create_graph, need_graph=True)
-
-    outputs = func(*inputs)
-    is_outputs_tuple, outputs = _as_tuple(outputs, "outputs of the user-provided function", "vjp")
-    _check_requires_grad(outputs, "outputs", strict=strict)
-
-    if v is not None:
-        _, v = _as_tuple(v, "v", "vjp")
-        v = _grad_preprocess(v, create_graph=create_graph, need_graph=False)
-        _validate_v(v, outputs, is_outputs_tuple)
+    with torch.enable_grad():
+        is_inputs_tuple, inputs = _as_tuple(inputs, "inputs", "vjp")
+        inputs = _grad_preprocess(inputs, create_graph=create_graph, need_graph=True)
+
+        outputs = func(*inputs)
+        is_outputs_tuple, outputs = _as_tuple(outputs, "outputs of the user-provided function", "vjp")
+        _check_requires_grad(outputs, "outputs", strict=strict)
+
+        if v is not None:
+            _, v = _as_tuple(v, "v", "vjp")
+            v = _grad_preprocess(v, create_graph=create_graph, need_graph=False)
+            _validate_v(v, outputs, is_outputs_tuple)
+        else:
+            if len(outputs) != 1 or outputs[0].nelement() != 1:
+                raise RuntimeError("The vector v can only be None if the "
+                                   "user-provided function returns "
+                                   "a single Tensor with a single element.")
+
+    if create_graph:
```

Comment:
I think you missed the proposed update here no?
In particular, you can 
```python
with torch.enable_grad(create_graph):
        grad_res = _autograd_grad(outputs, inputs, v, create_graph=create_graph)
        vjp = _fill_in_zeros(grad_res, inputs, strict, create_graph, "back")
```

I think you can do this update for all functions below.

Reply:
Thanks for reminding :)